ffi.c

```diff
@@ -605,7 +610,7 @@ int cast2ptr(lua_State *L, int idx, void **ptr)
 		switch (type->size) { FLOAT_TYPE_LIST_(COMPLEX_CASE) } \
 	}
 #else
-# define COMPLEX_CASES
+# define COMPLEX_CASES }
```

Comment:
I think the closing brace on this line (and on line 611) should really be on line 623 to close the switch statement.  This was probably a mistake when I factored the switch cases into macros.

Reply:
if FFI_TARGET_HAS_COMPLEX_TYPE is not defined, the switch statement will not close.

Reply:
I don't think the code will look strange.  Anyway, I have fixed the closing brace.

Reply:
you are right, that's a good place. thank you :D